### PR TITLE
D add log line file and func

### PR DESCRIPTION
**Pull request by vongang**
log into file
screen shot 2017-06-06 at 5 44 21 pm
std log to TTY
screen shot 2017-06-06 at 5 11 23 pm

### CODE REVIEW

File: log.go [hunk 1]
```diff
@@ -4,18 +4,53 @@ import (
...
+	return nil
+}
+
+func init() {
+	//log file, func and line in dep env
+	if Config.Env() == "production" {
+		// Log as JSON instead of the default ASCII formatter.
+		logger.SetFormatter(&logger.JSONFormatter{})
+		logger.SetOutput(os.Stderr)
+		logger.SetLevel(logger.InfoLevel)
```
Comment (vaporz) [inline log.go]: warn

File: config.go [hunk 2]
```diff
@@ -42,6 +42,10 @@ type config struct {
 	fieldMappings  map[string][]string
 }
 
+func (c *config) Env() string {
```
Comment (vaporz) [inline config.go]: add UT

File: log.go [hunk 3]
```diff
@@ -4,18 +4,53 @@ import (
...
 var log *logger.Logger
 
-func init() {
-	// Log as JSON instead of the default ASCII formatter.
-	logger.SetFormatter(&logger.TextFormatter{})
-	// Output to stdout instead of the default stderr
-	logger.SetOutput(os.Stdout)
-	//Log the Debug level and above.
-	logger.SetLevel(logger.DebugLevel)
+type ContextHook struct{}
```
Comment (vaporz) [inline log.go]: Comment

File: log.go [hunk 4]
```diff
@@ -4,18 +4,53 @@ import (
...
+
+func (hook ContextHook) Levels() []logger.Level {
```
Comment (vaporz) [inline log.go]: Comment

File: log.go [hunk 5]
```diff
@@ -4,18 +4,53 @@ import (
...
+	return logger.AllLevels
+}
 
+func (hook ContextHook) Fire(entry *logger.Entry) error {
```
Comment (vaporz) [inline log.go]: Comment, what does Fire() do?

**Comment by vaporz** [comment]
Please add support for logging to a file, and the file path should be configurable.

**Comment by vongang** [comment]
done

File: config.go [hunk 6]
```diff
@@ -36,12 +37,64 @@ var ServiceRootPath string
 // ServicePkgPath is the package path, e.g. "github.com/vaporz/turbo"
 var ServicePkgPath string
 
+// TurboRootPath is the absolute path to turbo
+
```
Comment (vaporz) [inline config.go]: remove line, use "gofmt -w ."

File: config.go [hunk 7]
```diff
@@ -36,12 +37,64 @@ var ServiceRootPath string
...
+var TurboRootPath string
+
 type config struct {
 	configs        map[string]string
 	urlServiceMaps [][3]string
 	fieldMappings  map[string][]string
 }
 
+//global configuration initializer
+
```
Comment (vaporz) [inline config.go]: remove line

File: config.go [hunk 8]
```diff
@@ -36,12 +37,64 @@ var ServiceRootPath string
...
+}
+
+//load turbo environment configuration
+func loadEnvConfig() {
+	env.configs = viper.GetStringMapString("config")
+}
+
+//init turbo basic env path, including RootPath and GOPATH
+func initPath() {
+	goPath := os.Getenv("GOPATH")
```
Comment (vaporz) [inline config.go]: duplicated codes with initPkgPath()

File: config.go [hunk 9]
```diff
@@ -36,12 +37,64 @@ var ServiceRootPath string
...
+
 type config struct {
 	configs        map[string]string
 	urlServiceMaps [][3]string
 	fieldMappings  map[string][]string
 }
 
+//global configuration initializer
+
+var env = &config{}
```
Comment (vaporz) [inline config.go]: create a new type for envConfig

File: config.go [hunk 10]
```diff
@@ -36,12 +37,64 @@ var ServiceRootPath string
...
+//init turbo environment
+func initEnv() {
+	initPath()
+	initViper()
+	loadEnvConfig()
+}
+
+//load turbo environment configuration
+func loadEnvConfig() {
+	env.configs = viper.GetStringMapString("config")
```
Comment (vaporz) [inline config.go]: tested? Will it conflict with the "config" node in service.yaml?

File: env.yml [hunk 11]
```diff
@@ -0,0 +1,3 @@
+config:
```
Comment (vaporz) [inline env.yml]: env.yml -> env.yaml

File: config.go (hunk 10 above)
Comment (vongang) [inline config.go]: yeah, tested. There is no conflict.

File: config.go (hunk 8 above)
Comment (vongang) [inline config.go]: done!

**Pull request merged**